packages/amplify-category-auth/provider-utils/awscloudformation/triggers/VerifyAuthChallengeResponse/function-template-dir/trigger-index.js

```diff
@@ -1,14 +1,58 @@
-/*
-  this file will loop through all js modules which are uploaded to the lambda resource,
-  provided that the file names (without extension) are included in the "MODULES" env variable.
-  "MODULES" is a comma-delimmited string.
-*/
+/**
+ * @fileoverview
+ *
+ * This CloudFormation Trigger creates a Lambda function which waits for all
+ * other specified Lambdas (modules) to resolve, which should be located in the
+ * same directory as this file (./).
+ *
+ * IMPORTANT: This handler *must* return the `event` since it may be consumed by
+ * additional layers in a CloudFormation stack. Failing to return an acceptable
+ * event may cause errors like "Invalid JSON" and prevent properly configured
+ * projects from deploying as expected.
+ *
+ * @see https://github.com/aws-amplify/amplify-cli/pull/7219
+ */
+
+/**
+ * The names of modules to load are stored as a comma-delimited string in the
+ * `MODULES` env variable.
+ */
 const moduleNames = process.env.MODULES.split(',');
+/**
+ * The array of imported modules.
+ */
 const modules = moduleNames.map(name => require(`./${name}`));
 
-exports.handler = (event, context, callback) => {
-  for (let i = 0; i < modules.length; i += 1) {
-    const { handler } = modules[i];
-    handler(event, context, callback);
-  }
+/**
+ * This handler is itself a Lambda function which iterates over all of the given
+ * modules and awaits them.
+ *
+ * @see https://docs.aws.amazon.com/lambda/latest/dg/nodejs-handler.html
+ *
+ * @param {object} event
+ *
+ * The event that triggered this Lambda.
+ *
+ * @param {object} context
+ *
+ * The context for this Lambda. See:
+ * https://docs.aws.amazon.com/lambda/latest/dg/nodejs-context.html
+ *
+ * @param {function} callback
+ *
+ * A deprecated way to send a response synchronously. See:
```

Comment:
I don't think we should use "synchronously" here. The linked documentation refers to them as "non-async."